k-distribution/include/builtin/domains.k

```diff
@@ -860,9 +861,16 @@ module K-REFLECTION-SYMBOLIC [symbolic]
   syntax Bool ::= #isVariable(K) [function, hook(KREFLECTION.isVariable)]
 endmodule
 
+module META
+  imports STRING-SYNTAX
+  // Takes as input a string and returns a K term
+  syntax KItem ::= #parseAST(String) [function, hook(META.parseAST), poly(0)]
```

Comment:
let's call this #parseKAST since it operates on KAST.

Reply:
I was thinking the same hook could be implemented in the haskell-backend to take as input KORE. Then parseAST would be a better name. But I don't feel strongly about it. I don't know what are the plans for the haskell-backend.

Reply:
I can say with 100% certainty that it makes no sense to have a hook that takes the same string to different outputs on different backends.